libz-rs-sys/src/gz.rs

```diff
@@ -238,10 +260,12 @@ unsafe fn gzopen_help(path: *const c_char, fd: c_int, mode: *const c_char) -> gz
 //
 // The caller must not use the state after passing it to this function.
 unsafe fn free_state(state: &mut GzState) {
-    if !state.path.is_null() {
-        ALLOCATOR.deallocate::<c_char>(state.path.cast_mut(), libc::strlen(state.path) + 1);
+    unsafe {
```

Comment:
here we just promise to only deallocate values that have been allocated with `ALLOCATOR`